walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -76,8 +71,58 @@
         }
       }
     }
+  }
+
+  private Collection<TimeRange> getAvailableTimesWithMaxOptional(Collection<TimeRange> availableTimes, Collection<Event> events, Collection<String> attendees, long duration){
+    if (attendees.isEmpty()){
+      return availableTimes;
+    }
+    // Overview: There's an ArrayList containing HashSets. All TimeRanges start off in the 
+    // HashSet in the first index. For each event/attendee combination, if it doesn't conflict
+    // with a TimeRange, move that TimeRange up an index. If it partially conflicts, split
+    // the TimeRange and move up the non-conflicting TimeRange. 
+    int size = attendees.size()*events.size() + 1;
+    ArrayList<HashSet<TimeRange>> rankNumOptionalCanAttend = new ArrayList<>(size);
+    rankNumOptionalCanAttend.add(new HashSet<>(availableTimes));
+    for (int i = 1; i < size; i++){
+      rankNumOptionalCanAttend.add(new HashSet<>());
+    }
+    for (String attendee: attendees){
+      for (Event event: events){
```

Comment:
It doesn't matter yet, but I changed the order in the second one because I wanted to change the algorithm to only consider attendees and not attendee/event pairs (because of the issue I described in a previous comment), but I didn't actually modify the algorithm yet.